src/main/java/hillel/spring/doctors/DoctorRestRepository.java

```diff
@@ -0,0 +1,73 @@
+package hillel.spring.doctors;
+
+
+import hillel.spring.doctors.model.Doctor;
+import lombok.AllArgsConstructor;
+import org.springframework.context.annotation.Bean;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+
+@Repository
+@AllArgsConstructor
+public class DoctorRestRepository {
+
+    private final List<Doctor> doctors = new CopyOnWriteArrayList<>();
+    private static int id;
+
+    @Bean
+    public AtomicInteger getAtomicInteger() {
+        return new AtomicInteger(doctors.size());
+    }
+
+    {
+        doctors.add(new Doctor(1, "Amosov", "cardiologist"));
+        doctors.add(new Doctor(2, "Pirogovskiy", "surgeon"));
+        doctors.add(new Doctor(3, "Dr. Sklifasovskiy", "surgeon"));
+
+        id = doctors.size();
+    }
+
+    public List<Doctor> findAllDoctors() {
+        return doctors;
+    }
+
+    public Integer createDoctor(Doctor doctor) {
+        id++;
```

Comment:
`++` is not an atomic operation. It consists of 3 ones: read, increment, write. So this ++ could be interrupted by another thread in between any of those. 
Why not use that atomic integer instead?

Reply:
Im using CopyOnWriteArrayList. It is thread safe under hood. So method create doctor is thread safe, i suppose